public/app/core/components/TimeSeries/utils.ts

```diff
@@ -497,7 +497,7 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn = ({
       barMaxWidth: customConfig.barMaxWidth,
       pointSize: customConfig.pointSize,
       spanNulls: customConfig.spanNulls || false,
-      show: !customConfig.hideFrom?.viz,
+      show: !customConfig.hideFrom?.viz && !field.state?.hideFrom?.viz,
```

Comment:
this `show` property is a hack that're really only needed for PlotLegend (i think).

we should actually never have a `show: false` uPlot series and omit adding any series that is not visible, instead of always adding a series with different `show` values.